### ISSUE TITLE
Convert to Tensorflow model. How to deal with Interpolation layer?

### ISSUE THREAD

**Issue by Vladkryvoruchko** [hszhao/PSPNet#45]
I want to try run this model in tensorflow, and is using 
[caffe-tensorflow](https://github.com/ethereon/caffe-tensorflow) converter.

There was some minor error which was fixed, but the main trouble is the interpolation layer.
If i am right - this is the custom layer made specifically for the semantic segmentation networks, and is absent in the master branch of caffe, so its not implemented in any converters.

Has anyone tried to convert it? And how did you face this problem?

**Comment by aneesh3108** [comment]
Hi @Vladkryvoruchko , 
I've been trying the same but haven't really got past an error telling me that "bn_param" is not a part of Caffe. 

Can you possibly share your current conversion output?? 
I'm working on extending the interpolation layer from DeepLab to tensorflow, but the current code seems to be a bottleneck for me.

Thanks!

**Comment by Vladkryvoruchko** [comment]
i'm fine with bn_param and caffe works well. But earlier i have made some corrections to prototxt 
![selection_021](https://user-images.githubusercontent.com/18577874/27075854-1963b1b4-5034-11e7-892f-294d01d92a20.png)

I'm working now with Keras and things seems to be fine, but havent finished yet

**Comment by Vladkryvoruchko** [comment]
@aneesh3108 
Please take a look at my results
`https://github.com/Vladkryvoruchko/PSPNet-Keras-tensorflow`

**Comment by aneesh3108** [comment]
@Vladkryvoruchko  On it this coming week. 
This is a great starter.

**Comment by amanjhunjhunwala** [comment]
Hey guys, since I couldnt find a tensorflow implementation took pudae's repo and added relevant stuff in it : (Accuracy obtained is the lower limit stated in the paper, but within range)--probably because I did not add an Auxillary Loss Layer !
If it is still useful : 
https://github.com/amanjhunjhunwala/PSPNet-Tensorflow

**Comment by wtliao** [comment]
@amanjhunjhunwala Hi, thanks for sharing your code. But the link seems inaccessible. Furthermore, do you have any idea how to add an Auxillary loss layer?

**Comment by amanjhunjhunwala** [comment]
@wtliao Due to certain reasons, I needed to close-source the code ! Sorry for the inconvenience !

**Comment by jaypatravali** [comment]
@Vladkryvoruchko Hey why is frozen=true in your prototxt, is this deploy. Just checking!
